cmd/ncproxy/ncproxy.go

```diff
@@ -24,8 +24,42 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+type computeAgentCache struct {
+	// lock for synchronizing read/write access to `cache`
+	rw *sync.RWMutex
```

Comment:
This will be `nil` by default and cause a panic when you try to lock it. We don't need it to be a pointer, so just change it to a value and it should work fine.

Reply:
This also points to the need for tests on this code. If we don't have anything automated, please make sure you're testing before sending PRs when possible.

Reply:
good catch, thank you. Unfortunately, we still don't have a way to test out ncproxy in an automated way from our perspective, but I will work on that for future PRs.

Reply:
Can we mock out the GRPC/TTRPC interfaces and do unit testing that way?

Reply:
Yea as long as the computeAgentCache functions take a pointer receiver (so the lock isn't copied) we should be fine

Reply:
Do you have something in mind for how those unit tests would look? Do you mean mock out the interfaces themselves or the calls in the implementation of the interfaces (iow mock `RegisterComputeAgent` vs mocking `winio.DialPipe`)?

Reply:
I was thinking we could provide mocks of the interfaces defined in the `.pb.go` files, but I think it's worth you spending time thinking about what testing we need for ncproxy.